newIDE/app/src/SceneEditor/index.js

```diff
@@ -386,7 +402,31 @@ export default class SceneEditor extends Component {
     done(true);
   };
 
-  _onRenameObject = (objectWithContext, newName, done) => {
+  //use this function to check if the new name can be used
+  // on apply new name event
+  _canObjectUseNewName = (objectWithContext, newName) => {
+    const { project, layout } = this.props;
+    const { object } = objectWithContext;
+
+    if (object.getName() === newName) {
+      return true;
+    } else if (
+      layout.hasObjectNamed(newName) ||
+      project.hasObjectNamed(newName)
+    ) {
+      showWarningBox('Another object with this name already exists.');
+      return false;
+    }
+    return true;
+  };
+
+  _onRenameEditedObject = newName => {
+    if (newName !== null) {
+      this._onRenameObject(this.state.editedObjectWithContext, newName);
```

Comment:
If you don't have anything to do after the object is renamed, just pass an empty function :)
```
this._onRenameObject(this.state. editedObjectWithContext, newName, () => {});
```

Reply:
Yes! Thats exactly what I was looking for! I wasn't sure how to pass an empty function as a parameter in javascript. Thank you! :D